packages/react/src/Avatar/Avatar.tsx

```diff
@@ -54,8 +60,67 @@ const Avatar = React.forwardRef<HTMLImageElement, AvatarProps>(function Avatar(
         sxProp as SxProp,
       )
     : merge({'--avatar-size': `${size}px`} as React.CSSProperties, sxProp as SxProp)
+
+  if (enabled) {
+    const cssSizeVars = {} as React.CSSProperties
+
+    if (isResponsive) {
+      for (const [key, value] of Object.entries(size)) {
+        // @ts-ignore - css property
+        cssSizeVars[`--avatarSize-${key}`] = `${value}px`
+      }
+    } else {
+      // @ts-ignore - css property
+      cssSizeVars['--avatarSize-regular'] = `${size}px`
+    }
```

Comment:
This is the biggest weirdness I was doing, trying to set CSS vars on the object if multiple sizes were passed in